src/modules/template-details/fork-template/ForkTemplate.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState, useCallback, useEffect } from 'react';
+
+import { Modal, Loader, Button } from 'ui';
+
+import { forkTemplate, Template } from 'core/api';
+
+import csx from './ForkTemplate.scss';
+
+export namespace ForkTemplate {
+  export interface Props {
+    template: Template;
+    onClose(): void;
+  }
+}
+
+interface State {
+  error: boolean;
+  pending: boolean;
+  success: boolean;
+}
+
+const STATE: State = {
+  error: false,
+  pending: true,
+  success: false
+};
+
+const ForkTemplate = ({ template, onClose }: ForkTemplate.Props) => {
+  const [state, setState] = useState<State>(STATE);
+
+  const { error, pending, success } = state;
+
+  const handleForkTemplate = useCallback(async (): Promise<void> => {
+    if (!pending) {
+      setState(STATE);
+    }
+
+    try {
+      await forkTemplate({ templateId: template.id });
+
+      setState({ error: false, pending: false, success: true });
+    } catch {
+      setState({ error: true, pending: false, success: false });
+    }
+  }, [pending]);
+
+  useEffect(() => {
+    handleForkTemplate();
+  }, []);
+
+  return (
+    <Modal className={csx.forkTemplate}>
+      {pending ? (
+        <Loader />
+      ) : error ? (
+        <>
```

Comment:
It would be nice to add some kind of very simple modal here - additional layer with background rgba with alpha set to e.g. 0.5/0.6, because now the dialog box is opened in the view center and it sometimes might overlap with other elements as they have similar background colors (in my case it doesn't look user experience friendly).

Reply:
This should be done in separate task